src/node_process_methods.cc

```diff
@@ -454,6 +425,83 @@ static void ReallyExit(const FunctionCallbackInfo<Value>& args) {
   env->Exit(code);
 }
 
+class FastHrtime {
+ public:
+  explicit FastHrtime(std::shared_ptr<v8::BackingStore> backing_store)
+      : backing_store_(backing_store) {}
+
+  static Local<Object> New(Environment* env) {
+    auto create_func = [env](auto fast_func, auto slow_func) {
+      auto cfunc = v8::CFunction::Make(fast_func);
+      return v8::FunctionTemplate::New(env->isolate(),
+                                       slow_func,
+                                       Local<Value>(),
+                                       Local<v8::Signature>(),
+                                       0,
+                                       v8::ConstructorBehavior::kThrow,
+                                       v8::SideEffectType::kHasNoSideEffect,
+                                       &cfunc);
+    };
+
+    auto number_tmpl = create_func(FastNumber, SlowNumber);
+    auto bigint_tmpl = create_func(FastBigInt, SlowBigInt);
+
+    Local<v8::ObjectTemplate> otmpl = v8::ObjectTemplate::New(env->isolate());
+    otmpl->SetInternalFieldCount(1);
+    otmpl->Set(FIXED_ONE_BYTE_STRING(env->isolate(), "hrtime"), number_tmpl);
+    otmpl->Set(FIXED_ONE_BYTE_STRING(env->isolate(), "hrtimeBigInt"),
+               bigint_tmpl);
+    Local<Object> obj = otmpl->NewInstance(env->context()).ToLocalChecked();
+
+    Local<ArrayBuffer> ab = ArrayBuffer::New(env->isolate(), 16);
```

Comment:
Should this be 12 bytes and not 16 since `hrtime()` only utilizes 3 32-bit integer values?

Reply:
the buffer is also used for bigint, so it needs to be a multiple of 8

Reply:
In that case it would just use the first 8 of 12 bytes and no more, correct?

Reply:
`new BigUint64Array(buffer)` validates the length.

Reply:
The length of the `BigUint64Array` instance is always 1, which uses 8 bytes. The length of the `Uint32Array` instance is always 3, which uses 12 bytes. Since the `ArrayBuffer` instance is shared between the two APIs, we only need to allocate 12 bytes to cover both APIs. The `BigInt` version will only use the first 8 bytes of the `ArrayBuffer` instance, while the other will use all 12 bytes.

Reply:
`new BigUint64Array(new ArrayBuffer(12))` will throw, it has to be a multiple of 8.

Reply:
Ah I see. Perhaps we could then do something like:

```js
const hrBigintValues = new BigUint64Array(_hrtime.buffer.subarray(0, 8));
```

Reply:
subarray is an operation on TypedArrays, not ArrayBuffers. afaik we have to just use 16. Technically this PR has actually reduced the amount of ArrayBuffer memory by four bytes since it combined the two ArrayBuffers.

Reply:
Ah I see. We should be able to then do something like:

```js
const hrBigintValues = new BigUint64Array(_hrtime.buffer, 0, 8);
```